python_modules/libraries/dagster-aws/dagster_aws/secretsmanager/resources.py

```diff
@@ -70,40 +108,127 @@ def example_job():
               # profile as specified in ~/.aws/credentials file
 
     """
-    return construct_secretsmanager_client(
-        max_attempts=context.resource_config["max_attempts"],
-        region_name=context.resource_config.get("region_name"),
-        profile_name=context.resource_config.get("profile_name"),
-    )
+    return SecretsManagerResource.from_resource_context(context).create_client()
 
 
-@resource(
-    merge_dicts(
-        BOTO3_SESSION_CONFIG,
-        {
-            "secrets": Field(
-                Array(str),
-                is_required=False,
-                default_value=[],
-                description="An array of AWS Secrets Manager secrets arns to fetch.",
-            ),
-            "secrets_tag": Field(
-                Noneable(str),
-                is_required=False,
-                default_value=None,
-                description=(
-                    "AWS Secrets Manager secrets with this tag will be fetched and made available."
-                ),
-            ),
-            "add_to_environment": Field(
-                bool,
-                is_required=False,
-                default_value=False,
-                description="Whether to mount the secrets as environment variables.",
-            ),
-        },
+class SecretsManagerSecretsResource(ResourceWithBoto3Configuration):
```

Comment:
i know this is based off of the old code, but this name is super confusing to me. I had to read the doc block 3 times to know what it was. if a rename is feasible, something more along the lines of `SpecificSecretsManagerResource` makes more intuitive sense to me, even though `SpecificSecretsSecretsManager` might be more accurate